crates/circuit/src/dag_circuit.rs

```diff
@@ -171,6 +171,23 @@ impl Wire {
     }
 }
 
+/// Allows the acceptance of either owned bit indices or a borrowed slice.
+pub enum OwnedOrSlice<'a, T> {
+    Slice(&'a [T]),
+    Owned(Vec<T>),
+}
+impl<'a, T> From<&'a [T]> for OwnedOrSlice<'a, T> {
+    fn from(value: &'a [T]) -> Self {
+        Self::Slice(value)
+    }
+}
+
+impl<'a, T> From<Vec<T>> for OwnedOrSlice<'a, T> {
+    fn from(value: Vec<T>) -> Self {
+        Self::Owned(value)
+    }
+}
```

Comment:
If we do decide to keep the functionality, this enum is actually just a concrete case of something that exists in the standard library already, and will play nicely with the interners: `std::borrow::Cow<'a, [T]>`.

Reply:
Oh, I didn't know this existed already. That's a relief!